Add md5sum check

squiddle opened the pull request:
this is a cleaned up version of https://github.com/maestrodev/puppet-wget/pull/7

carlossg commented on manifests/fetch.pp:
```diff
@@ -110,6 +112,16 @@
 
+  case $sourceHash {
+    '', undef: {
+      $command = "wget ${verbose_option}${nocheckcert_option}${no_cookies_option}${header_option}${user_option}${output_option}${flags_joined} '${source}'"
```
$command is never used

carlossg commented on manifests/fetchifhashchanged.pp:
```diff
@@ -0,0 +1,35 @@
+  $verbose          = false,
+) {
+
+  exec { "rm ${destination} ; echo ignoredresult":
```
should just be `rm -f` ?

carlossg commented:
I've added some comments. It'd be good to

* add unit and system tests
* squash the commits into 2: one for implementing retries and another for the sourceHash
* add docs

I would implement the hash change into the same definition instead of creating a new one

squiddle commented:
- i added unit and system tests
- i removed the retries change, as i do not need it, it was just part of the original pull request
- added additional documentation for $source_hash
- the removal of $destination if $source_hash does not match is now always done, i could not think of why puppet should keep a file, where it knows it is not what is inteded.
- first time i used rebase and push --force for already published changes, feels exciting  :) i hope this is correct now

squiddle commented:
- i also fixed the $command issue you mentioned
- and i changed the removal operation to use `test`; feels better than a forced rm
